src/common/workspace-store.ts

```diff
@@ -310,7 +310,7 @@ export class WorkspaceStore extends BaseStore<WorkspaceStoreModel> {
     }
 
     const currentWorkspaces = this.workspaces.toJS();
-    const newWorkspaceIds = new Set<WorkspaceId>();
+    const newWorkspaceIds = new Set<WorkspaceId>([WorkspaceStore.defaultId]); // never delete default
```

Comment:
oh yeah, that's important. :)
You could hold changes to this file till 4.2.1 (the other changes fix the problem Mario found, right?)

Reply:
I think so, though I think he was also running into this so I think I'll keep it. I'll open another PR about the other stuff which can be left to 4.2.1

Reply:
Tests would be nice 🙂

Reply:
Tests caught this change, do you mean more tests?

Reply:
I meant more tests (I don't see any added in this PR).